services/softlayer_account_test.go

```diff
@@ -96,6 +96,19 @@ var _ = Describe("SoftLayer_Account_Service", func() {
 		})
 	})
 
+	Context("#GetIscsiNetworkStorageWithFilter", func() {
+		BeforeEach(func() {
+			fakeClient.DoRawHttpRequestResponse, err = common.ReadJsonTestFixtures("services", "SoftLayer_Account_Service_getNetworkStorage.json")
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("returns an array of datatypes.SoftLayer_Network_Storage", func() {
+			iscsiNetworkStorage, err := accountService.GetIscsiNetworkStorageWithFilter("fake-filter")
```

Comment:
In fact, filters would not be checked inside the called method, the filter would like to be constructed in the callingmethod, so if a invalid filter was passed into `GetIscsiNetworkStorageWithFilter`, the called method (`GetIscsiNetworkStorageWithFilter`) would like to return error, and calling method could catch and handle appropriately.